Add Linux to platform channel example (#8127)
flutter/website: src/development/platform-integration/platform-channels.md

| Map                        | FlValue(FlValue, FlValue) |
{% endsamplecode %}

> ## Example: Calling platform-specific Android, iOS, Windows, and Linux code using platform channels {#example}

Reviewer: This heading is getting really long. Should we simplify it?

```suggestion
## Example: Calling platform-specific code using platform channels {#example}
```

Author: SGTM

You should now be able to run the application on Windows.
If your device doesn't have a battery,
it displays 'Battery level not available'.
  
### Step 6: Add a Linux platform-specific implementation
  

> For this example you will need the `upower` developer headers installed.

Reviewer 2: ```suggestion
For this example you need to install the `upower` developer headers.
```

This is likely available from your distribution, for example with:
```sh
sudo apt install libupower-glib-dev
```

Start by opening the Linux host portion of your Flutter app in the editor
of your choice. The instructions below are for Visual Studio Code with the
“C/C++” and “CMake” extensions installed, but can be adjusted for other IDEs.

1. Launch Visual Studio Code.

1. Open the **linux** directory inside your project.

1. Choose **Yes** in the prompt asking: `Would you like to configure project "linux"?`.

>    This will allow C++ autocomplete to work.

Reviewer 2: ```suggestion
   This enables C++ autocomplete.
```

1. Open the file `my_application.cc`.
  
First, add the necessary includes to the top of the file, just
after `#include <flutter_linux/flutter_linux.h`:

<!--code-excerpt "my_application.cc" title-->
```c
#include <math.h>
#include <upower.h>
```

Add an `FlMethodChannel` to the `_MyApplication` struct:

<!--code-excerpt "my_application.cc" title-->
```c
struct _MyApplication {
  GtkApplication parent_instance;
  char** dart_entrypoint_arguments;
  FlMethodChannel* battery_channel;
};
```

> Make sure it is cleaned up in `my_application_dispose`

Reviewer 2: ```suggestion
Make sure to clean it up in `my_application_dispose`:
```

<!--code-excerpt "my_application.cc" title-->
```c
static void my_application_dispose(GObject* object) {
  MyApplication* self = MY_APPLICATION(object);
  g_clear_pointer(&self->dart_entrypoint_arguments, g_strfreev);
  g_clear_object(&self->battery_channel);
  G_OBJECT_CLASS(my_application_parent_class)->dispose(object);
}
```

> Edit the `my_application_activate` method and initilize

Reviewer: ```suggestion
Edit the `my_application_activate` method and initialize
```
